tests/jballerina-unit-test/src/test/resources/test-src/types/tuples/tuple_rest_descriptor_test.bal

```diff
@@ -178,6 +178,30 @@ function testSubTypingWithRestDescriptorNegative() {
     assertFalse(v2 is [string, string, string, string]);
 }
 
+function testRestVariablesWithArray() {
+    int[] [...a] = [1, 2];
+    int[] _ = a;
+    assertEquality(a, <int[]>[1, 2]);
```

Comment:
Why do we need this cast?

Reply:
It is because to verify that the type is exactly same. [1, 2] is a tuple in default I guess, So test will not pass.

Reply:
Please refer to this thread. https://github.com/ballerina-platform/ballerina-lang/pull/37784#discussion_r976019099